### PR TITLE
refactor(overflow-menu): update examples to match React integration

### LINKED ISSUE #2304: Overflow-menu updates
**Placeholder issue for overflow-menu udpates**

- add `hidden` to `.pf-c-overflow-menu__control`
- add class `.pf-m-hidden` to `__items/__groups` rather than relying on `pf-m-shared`

### PR DESCRIPTION
**Pull request by mattnolting**
closes #2304 

- removes modifiers where JS handles appearance
- adds non-responsive shown/hidden examples

### CODE REVIEW

File: src/patternfly/components/OverflowMenu/docs/code.md [hunk 1]
```diff
@@ -1,21 +1,12 @@
...
+| `.pf-c-overflow-menu__item` | `<div>` | Initiates an overflow menu item. **Required** |
 | `.pf-m-button-group` | `.pf-c-overflow-menu__group` | Modifies overflow menu group spacing. Spacer value is set to `var(--pf-c-overflow-menu__group--m-button-group--spacer)`. Child spacer value is set to `var(--pf-c-overflow-menu__group--m-button-group--space-items)`. |
 | `.pf-m-icon-button-group` | `.pf-c-overflow-menu__group` | Modifies overflow menu group spacing. Spacer value is set to `var(--pf-c-overflow-menu__group--m-icon-button-group--spacer)`. Child spacer value is set to `var(--pf-c-overflow-menu__group--m-icon-button-group--space-items)`. |
-| `.pf-m-additional-options` | `.pf-c-overflow-menu__control` | Modifies overflow menu control visibility. |
-| `.pf-m-persistent` | `.pf-c-overflow-menu__item`, `.pf-c-overflow-menu__group` | Modifies overflow menu group/item visibility. |
-| `.pf-m-show{-on-[md, lg, xl]}` | `.pf-c-overflow-menu` |  Specifies at which breakpoint overflow menu items and groups are shown. **Required** |
+| `.pf-m-show{-on-[md, lg, xl]}` | `.pf-c-overflow-menu` | Modifies overflow element spacing at specified breakpoint. **Required** |
```
Comment (mcoker) [inline src/patternfly/components/OverflowMenu/docs/code.md]: is that the right description?

File: src/patternfly/components/OverflowMenu/overflow-menu-dropdown-item.hbs [hunk 2]
```diff
@@ -1,6 +1,6 @@
-<li class="{{#if overflow-menu-dropdown-item--modifier}}{{overflow-menu-dropdown-item--modifier}}{{/if}}{{#if overflow-menu-dropdown-item--IsShared}} pf-c-overflow-menu__shared-item{{/if}}"
+<li {{#if overflow-menu-dropdown-item--modifier}}class="{{overflow-menu-dropdown-item--modifier}}"{{/if}}
```
Comment (mcoker) [inline src/patternfly/components/OverflowMenu/overflow-menu-dropdown-item.hbs]:
```suggestion
<li{{#if overflow-menu-dropdown-item--modifier}} class="{{overflow-menu-dropdown-item--modifier}}"{{/if}}
```

File: src/patternfly/components/OverflowMenu/docs/code.md (hunk 1 above)
Comment (mattnolting) [reply src/patternfly/components/OverflowMenu/docs/code.md]: nope, updating

**Comment by mattnolting** [comment]
@mcoker removed references to `pf-m-show`

**Comment by mattnolting** [comment]
@mcoker @christiemolloy updated

File: src/patternfly/components/OverflowMenu/overflow-menu.scss [hunk 3]
```diff
@@ -27,21 +18,23 @@ $pf-c-overflow-menu--breakpoint-map: build-breakpoint-map("base", "md", "lg", "x
...
-  --pf-c-overflow-menu__control--Display: inline-block;
-  --pf-c-overflow-menu__control--Visibility: visible;
-
   display: inline-flex;
   align-items: center;
 }
 
+// Content
+.pf-c-overflow-menu__content {
+  // reset spacer
```
Comment (mcoker) [inline src/patternfly/components/OverflowMenu/overflow-menu.scss]:
```suggestion
```

File: src/patternfly/components/OverflowMenu/examples/OverflowMenu.md [hunk 4]
```diff
@@ -105,56 +123,51 @@ The overflow menu relies on groups (`.pf-c-overflow-menu__group`) and items (`.p
...
-      Item 3
-    {{/overflow-menu-dropdown-item}}
-    {{#> overflow-menu-dropdown-item overflow-menu-dropdown-item--IsShared="true"}}
+The action group consists of a primary and secondary action. Any additional actions are part of the overflow control dropdown.
+
+### Overflow menu group types
+
+| Class | Applied to | Outcome |
+| -- | -- | -- |
+| `.pf-c-overflow-menu__group` | `<div>` | Initiates the overflow menu component group. |
```
Comment (mcoker) [inline src/patternfly/components/OverflowMenu/examples/OverflowMenu.md]:
```suggestion
| `.pf-c-overflow-menu__group` | `<div>` | Initiates an overflow menu component group. |
```

**Pull request merged**